### ISSUE TITLE
Suggestion: Remove sklearn as a dependency

### ISSUE THREAD

**Issue by qiyunzhu** [biocore/scikit-bio#1827]
I noticed that scikit-learn is a dependency of scikit-bio. However, the only place it is used is in the calculation of [beta diversity](https://github.com/biocore/scikit-bio/blob/master/skbio/diversity/_driver.py) metrics. After examining the code, I believe that this heavy dependency can be totally removed, thus making scikit-bio slimmer, quicker to load, and more coherent to the original idea of `scikit` (meaning extensions of SciPy, if I understand correctly; thus I guess scikits probably shouldn't depend on each other).

Specifically, according to the code and [documentation](http://scikit-bio.org/docs/latest/generated/skbio.diversity.beta_diversity.html), it wraps any metric provided by scikit-learn's [`pairwise.distance_metrics`](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.distance_metrics.html). If not found, it will go for SciPy's [`pdist`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.pdist.html). Actually, scikit-learn only provides a handful of distance metrics, most of which are covered by SciPy. Some non-overlapping metrics are synonyms, such as: l1 = manhattan, l2 = euclidean. The only metric that scikit-learn has but SciPy does not is the [Haversine distance](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.haversine_distances.html), but it measures the distance between two points on a spheral surface, and I doubt if anyone would use it for calculating beta diversity between two communities.

Therefore I think the wrapper for scikit-learn metrics can be removed from scikit-bio, letting SciPy be the only source of (non-phylogenetic) beta diversity metrics. If needed (for sake of ecology; not for geometry) one can manually implement additional beta diversity metrics.

**Comment by ebolyen** [comment]
I think this is a good idea, the only catch is we will need to make sure that some of our qualitative metrics cast numeric vectors into boolean before running. This is something sklearn does behind the scenes, but it is really quite trivial and we should be handling it on our end anyhow.

**Comment by gregcaporaso** [comment]
We addressed this in #1831 - thanks for the suggestion @qiyunzhu!

**Comment by qiyunzhu** [comment]
@gregcaporaso @ebolyen Thanks! Looks great!
